python/hsfs/core/job_scheduler.py

```diff
@@ -45,45 +54,255 @@ def __init__(
+            "WEEKLY",
+        ]:
+            raise ValueError(
+                "job_frequency must be one of: 'HOURLY', 'DAILY', 'WEEKLY' or a cron_expression must be provided.\n"
+                + f" Got: {job_frequency}"
+            )
         self._id = id
-        self._start_date_time = util.convert_event_time_to_timestamp(start_date_time)
-        self._job_frequency = job_frequency
         self._job_name = job_name
         self._enabled = enabled
+        self._cron_expression = cron_expression
+        self.job_frequency = job_frequency
+        self.start_date_time = start_date_time
+        self.end_date_time = end_date_time
+        self._next_execution_date_time = util.convert_event_time_to_timestamp(
+            next_execution_date_time
+        )
+        self._strftime_format = "%Y-%m-%dT%H:%M:%S.%fZ"
+        self._job_scheduler_engine = job_scheduler_engine.JobSchedulerEngine()
 
     @classmethod
     def from_response_json(cls, json_dict):
         return cls(**humps.decamelize(json_dict))
 
-    def to_dict(self):
+    def save(self) -> "JobScheduler":
+        """Saves the scheduler configuration to Hopsworks.
+
+        # Returns
+            `JobScheduler` The scheduler configuration.
+        """
+        if self._id:
+            raise ValueError(
+                "Cannot save a scheduler already registered, use `update()` to persist edit to Hopsworks."
+            )
+
+        return self._job_scheduler_engine.create_job_scheduler(the_job_scheduler=self)
+
+    def delete(self) -> None:
+        """Deletes the scheduler configuration from Hopsworks.
+
+        !!! info
+            The job itself is not deleted, only the scheduler configuration.
+        """
+        if not self._id:
+            raise ValueError(
+                "Cannot delete a scheduler not registered, use `save()` to register a new scheduler to Hopsworks."
+            )
+        if not self._job_name:
+            raise ValueError("Cannot delete a scheduler without a job name.")
+
+        self._job_scheduler_engine.delete_job_scheduler(job_name=self._job_name)
+
+    def update(self) -> "JobScheduler":
+        """Updates the scheduler configuration in Hopsworks.
+
+        # Returns
+            `JobScheduler` The scheduler configuration.
+        """
+        if not self._id:
+            raise ValueError(
+                "Cannot update a scheduler not registered, use `save()` to register a new scheduler to Hopsworks."
+            )
+
+        return self._job_scheduler_engine.update_job_scheduler(the_job_scheduler=self)
+
+    def pause(self):
+        """Pauses the scheduling of job in Hopsworks, the job can still be triggered manually."""
+        if self._id:
+            raise ValueError(
+                "Cannot pause a scheduler not registered, use `save()` to register a new scheduler to Hopsworks."
+            )
+        if not self._job_name:
+            raise ValueError("Cannot pause a scheduler without a job name.")
```

Comment:
Could this be actually possible? An `existing` job scheduler without a job name?

Reply:
I am not sure but since the job name is basically used to query the rest api all the time I would say we keep a lot of checks in place to avoid issues